tests/components/influxdb/test_sensor.py

```diff
@@ -451,3 +451,59 @@ async def test_error_rendering_template(
     assert (
         len([record for record in caplog.records if record.levelname == "ERROR"]) == 1
     )
+
+
+@pytest.mark.parametrize(
+    "mock_client, config_ext, queries, set_query_mock, test_exception",
+    [
+        (
+            DEFAULT_API_VERSION,
+            BASE_V1_CONFIG,
+            BASE_V1_QUERY,
+            _set_query_mock_v1,
+            OSError("fail"),
+        ),
+        (
+            DEFAULT_API_VERSION,
+            BASE_V1_CONFIG,
+            BASE_V1_QUERY,
+            _set_query_mock_v1,
+            InfluxDBClientError("fail"),
+        ),
+        (
+            DEFAULT_API_VERSION,
+            BASE_V1_CONFIG,
+            BASE_V1_QUERY,
+            _set_query_mock_v1,
+            InfluxDBServerError("fail"),
+        ),
+        (
+            API_VERSION_2,
+            BASE_V2_CONFIG,
+            BASE_V2_QUERY,
+            _set_query_mock_v2,
+            OSError("fail"),
+        ),
+        (
+            API_VERSION_2,
+            BASE_V2_CONFIG,
+            BASE_V2_QUERY,
+            _set_query_mock_v2,
+            ApiException(),
+        ),
+    ],
+    indirect=["mock_client"],
+)
+async def test_connection_error_at_startup(
+    hass, caplog, mock_client, config_ext, queries, set_query_mock, test_exception
+):
+    """Test behavior of sensor when influx returns error."""
+    set_query_mock(mock_client, side_effect=test_exception)
+
+    with patch(f"{INFLUXDB_SENSOR_PATH}.event_helper") as event_helper:
+        await _setup(hass, config_ext, queries, [])
+        assert (
+            len([record for record in caplog.records if record.levelname == "ERROR"])
+            == 1
+        )
+        event_helper.call_later.assert_called_once()
```

Comment:
As mentioned in https://github.com/home-assistant/core/pull/37181, this was the one test I couldn't add prior to the refactor. The sensor setup was a bit different then the main component setup in that it didn't have this retry on failure logic. Now that it is added I can create the test for it here as well.

Reply:
We should `raise PlatformNotReady` in the sensor platform instead to have the set up platform retry.

Reply:
@MartinHjelmare how do you test this behavior? The `PlatformNotReady` exception is caught and handled by HA's platform setup logic. For this test I would want to know that setup had failed and this exception was being raised but wouldn't want to actually have it retry. Should I call the `setup_platform` method directly for this test then or is there a preferred way to do this?

Reply:
Ok adjusted the test to directly call `setup_platform` and assert that `PlatformNotReady` was raised. If there is a different best practice way to do this test let me know and I'll adjust

Reply:
Assert that there's no entity state on error, and then resolve the error and move time forward and assert that there's an entity state.

Use `async_fire_time_changed` from tests/common.py to move time forward.

Reply:
Ok all set